internal/querynode/task.go

```diff
@@ -656,13 +650,29 @@ func (l *loadSegmentsTask) Execute(ctx context.Context) error {
 		}
 	}
 
-	err = l.node.loader.loadSegment(l.req, segmentTypeSealed)
+	// filter segments that are already loaded in this querynode
+	var filteredInfos []*queryPb.SegmentLoadInfo
+	for _, info := range l.req.Infos {
+		if !l.node.historical.replica.hasSegment(info.SegmentID) {
+			filteredInfos = append(filteredInfos, info)
+		} else {
+			log.Debug("ignore segment that is already loaded", zap.Int64("segmentID", info.SegmentID))
```

Comment:
NIT: add more information about segment such as collectionID and partitionID for debugging.

Reply:
Will fix in the next PR